### PR TITLE
new en hyph patterns

### PR DESCRIPTION
**Pull request by cramoisi**
I reread LOTR and a bunch of Moore's Jerusalem with them and everything seems OK (no difference with the old pdb files but the file format itself is an improvement), so here are the English_UK and English_GB pattern files.
No need to change the pattern size (35 for german).
I set right and left min at 2 as these pattern are made for.
The PR comes with a setting for french at 2/1 as I made the pattern file for using it this way.
And I don't know why the path for c3gui/data was in `.gitignore` ; I removed it since we need access to hyph which is inside. Tell me if I'm wrong.

### CODE REVIEW

File: cr3gui/data/hyph/languages.json [hunk 1]
```diff
@@ -41,12 +41,16 @@
     },
     {
         "name": "English (UK)",
-        "filename": "English_GB_hyphen_(Alan).pdb",
+        "filename": "English_GB.pattern",
+	"left_hyphen_min": 2,
+	"right_hyphen_min":2,
```
Comment (poire-z) [inline cr3gui/data/hyph/languages.json]: You got some tabulation at start of those lines, rather replace them with 8 spaces like all the others.

**Comment by cramoisi** [comment]
tss, it's scite.

**Comment by poire-z** [comment]
> And I don't know why the path for c3gui/data was in .gitignore ; I removed it since we need access to hyph which is inside. Tell me if I'm wrong.

It was added in https://github.com/koreader/crengine/commit/eedb7ee110378207e69e78f4637d359a55bfb526
I personally don't mind removing it, as it was often hiding the work/backup files we may have make in c3gui/data/ (styles, hyph dict), and I prefer seeing my crap with `git status`. @Frenzie : any opinion?

**Comment by Frenzie** [comment]
Its exclusion does not seem relevant to our current situation.

**Comment by cramoisi** [comment]
There is also the possibility to ignore the other subdirectories of data.

**Comment by cramoisi** [comment]
just to note : I manage to get access to the content of the pdb files. The rules are structured like that : 
?string1RULE1?string2RULE2 where string is letters and PATTERN numbers as shown by crengine when hyph is in debug mode. So it's something like that : ?université0030040300. Not very interesting (since I don't know how to make one pdb file) but it was only to be sure there will be no drawback (as, for example, all the old patterns for english where included or updated in the new files) to switch for dot pattern. With that known I can easily proof check and convert the other even if I can't read the language at all.

**Comment by poire-z** [comment]
The reason for it to be in .gitignore was probably that without, these show now in git status:
```
        cr3gui/data/cr3.css
        cr3gui/data/cr3.ini
        cr3gui/data/dict/
```
I'd better re-add in next commit, no?

**Comment by Frenzie** [comment]
In that case I think it'd be better to ignore those more targeted. But why do they show?

**Comment by cramoisi** [comment]
we only need to add these : 
```
cr3gui/data/desktop/
cr3gui/data/devices/
cr3gui/data/manual/
cr3gui/data/cr3.css
cr3gui/data/cr3.ini
cr3gui/data/dict/
```
and : 

cr3gui/data/cr3.css
cr3gui/data/cr3.ini
cr3gui/data/dict/

show on git status because they are create on first run by the emulator and are symlink to the git tree (they should be in the emulator tree for the two first, and if koreader won't embark a dict, the third should be in the emulator tree too)

**Pull request merged**